SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -36,17 +38,31 @@ async def get(self):
             # The first get request will launch the execution of the makenv.sh script,
             # to create the environment
             makenv_process = self._launch_makenv()
-            SwanCustomEnvironmentsApiHandler.makenv_process = makenv_process
             log_file = open(self.LOG_FILE, "w")
+            SwanCustomEnvironmentsApiHandler.makenv_process = makenv_process
+            SwanCustomEnvironmentsApiHandler.writing_log = True
         else:
+            # Parent process has already been launched, so we will get its generated output
             ret_code = makenv_process.poll()
-            if ret_code is not None and ret_code != 0:
-                # If the process has finished with error, show the content of the log file
-                with open(self.LOG_FILE, "r") as log_file:
+            if ret_code is None or ret_code != 0:
+                log_file = open(self.LOG_FILE, "r")
+                if ret_code is None:
```

Comment:
This is a bit convoluted, you are checking again a condition you already checked. Do we need a different behaviour depending on whether the makenv process is still running or not, and whether it has finished with error or not? Don't we just need to output its log?